### PR TITLE
GH Issue: Set Numpy < 2.0.0 as a requirement since it breaks our package

### LINKED ISSUE #1680: ImportError: numpy.core.multiarray failed to import
# Describe the bug
ImportError with cv2 and numpy when using labelbox.
Stack trace:
```
    from labelbox.data.annotation_types import ImageData, Label
/usr/local/lib/python3.11/site-packages/labelbox/data/annotation_types/__init__.py:1: in <module>
    from .geometry import Line
/usr/local/lib/python3.11/site-packages/labelbox/data/annotation_types/geometry/__init__.py:1: in <module>
    from .line import Line
/usr/local/lib/python3.11/site-packages/labelbox/data/annotation_types/geometry/line.py:5: in <module>
    import cv2
E   ImportError: numpy.core.multiarray failed to import
```

Also:
```
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.
```

# To reproduce
1. Run `from labelbox.data.annotation_types import ImageData, Label` 

# Environment (please complete the following information
 - OS: macOS Sonoma 14.5
 - Python Version: 3.11.8
 - SDK + SDK Version: labelbox[data]==3.67.0

# Additional context
I believe the required fix should be to set the numpy optional dependency to `numpy>=1.24.4,<2.0`

### PR DESCRIPTION
**Pull request by Gabefire**
# Description

* From the linked issue, if someone has numpy >2.0, it breaks our package. This is due to Numpy >2,0 supporting Python 3.9 and above and 2.0 being a major release (first major release since 2006), i.e., changing a lot of how Numpy functions. Specified version 2.0 or below with rye to avoid confusion. Also cleaned up whitespace in pyproject file. See linked issue for more information

* [ Numpy](https://numpy.org/devdocs/release/2.0.0-notes.html) just released this version on the 6/13. Provided a comment to pyproject for information we can remove this once we removed support for 3.8 in Oct due to EOL of 3.8

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## All Submissions

- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you provided a description?
- [x] Are your changes properly formatted?

### CODE REVIEW

**Review by adrian-chang — DISMISSED** [review]
Need to rebuild requirements.lock

eg `rye sync` or `rye sync --update-all -f` if you want to try and update all deps

**Comment by Gabefire** [comment]
> Need to rebuild requirements.lock
> 
> eg `rye sync` or `rye sync --update-all -f` if you want to try and update all deps

@adrian-chang So I did a ton of research once you mentioned this. Here is a quick [loom ](https://www.loom.com/share/18953b025c934c2aba417915ad9303f7?sid=f7751651-ebe2-4173-a537-9f7f284e2421)
I did rye sync, but it did not change anything because I think it is due to these being an optional dependency. The current requirement don't have the optional dependencies. This loom also highlights the Pillow clean up fix I did. Please let me know what you think!

TLDR: essentially the optional dependencies don't get included on the requirements.lock. When install "labelbox[data]" it ~~gives a pillow install error (pinned this version)~~ and numpy 2.0.0 that was just released breaks the packageing, so pinned this version

Edit: disregard pillow change this is a Colab itself issue thus it is not a concern for us and should support the highest bug fix release

`rye lock --all-features` would include all the extra dependencies to the requirement.lock file if you want that

**Review by vbrodsky — APPROVED** [review]
